src/python/grpcio/grpc/experimental/aio/_typing.py

```diff
@@ -13,17 +13,18 @@
 # limitations under the License.
 """Common types for gRPC Async API"""
 
-from typing import (Any, AnyStr, AsyncIterable, Callable, Iterable, Sequence,
-                    Tuple, TypeVar, Union)
+from typing import (Any, AsyncIterable, Callable, Iterable, Sequence, Tuple,
+                    TypeVar, Union)
 
 from grpc._cython.cygrpc import EOF
+from ._metadata import Metadata, MetadataKey, MetadataValue
 
 RequestType = TypeVar('RequestType')
 ResponseType = TypeVar('ResponseType')
 SerializingFunction = Callable[[Any], bytes]
 DeserializingFunction = Callable[[bytes], Any]
-MetadatumType = Tuple[str, AnyStr]
-MetadataType = Sequence[MetadatumType]
+MetadatumType = Tuple[MetadataKey, MetadataValue]
+MetadataType = Metadata
```

Comment:
Should we keep the backward compatibility? Like `Union[Sequence[MetadatumType], Metadata]`.

Reply:
Maybe I'm missing something, but if I'm not wrong the `MetadataType` is only used internally or for returning a metadata object to the user, never for getting data from the user. If this is the case we wouldn't need to keep backward compatibility, indeed we could remove `MetadataType` and just annotate the attributes with a simple `Metadata`.

am I wrong?

Reply:
@pfreixes I agree, I'm replacing the ``MetadataType`` by ``Metadata``